lib/common/schemas.c

```diff
@@ -1275,3 +1275,30 @@ cli_config_update(xmlNode **xml, int *best_version, gboolean to_logs)
     free(orig_value);
     return rc;
 }
+
+/*!
+ * \internal
+ * \brief Return the contents of the schema file matching a given
+ *        schema version
+ *
+ * \param[in] name  The schema version to use (for example, "pacemaker-3.1")
+ *
+ * \note The caller is responsible for freeing the returned value
+ */
+char *
+pcmk__schema_contents(const char *name)
+{
+    int rc = pcmk_rc_ok;
+    char *contents = NULL;
+    char *path = pcmk__xml_artefact_path(pcmk__xml_artefact_ns_legacy_rng,
+                                         name);
```

Comment:
I'd kind of disregarded this approach at first because I thought `pcmk__xml_artefact_path` was used for constructing paths to write to in addition to constructing search paths.  That turns out to not be the case, so I've added PCMK_remote_schema_directory (which needs another patch to edit the sysconfig file, I'll do that in a sec).  This also allows me to drop the patch that keeps track of where schema files come from.